lib/eib.sh

```diff
@@ -271,12 +271,14 @@ sign_file() {
   fi
 }
 
-# Create a detached checksum with sha256sum.
+# Create a detached checksum with sha256sum. By default the target in
+# the checksum file is the basename of the file being checksummed.
 checksum_file() {
   local file=${1:?No file supplied to ${FUNCNAME}}
   local outfile=${2:-${file}.sha256}
+  local target=${3:-${file##*/}}
 
-  sha256sum "${file}" | cut -d' ' -f1 > "${outfile}"
+  sha256sum "${file}" | awk "{print \$1 \"  $target\"}" > "${outfile}"
```

Comment:
I don't quite understand why the `awk` step is needed – to strip the path down to a basename? Would another option be to `cd` to the directory the target is in in a subshell? Maybe it's no easier.

Reply:
Because the path you're using for the checksum is not necessarily the same as what you want in the file. It could be because you're not in the same directory or it could be because the file name doesn't match what you want at the time the command is run. That's what happens in the extracted split case and could be in the extracted image case, although the raw image temporary file is named like the output file.

Reply:
Would it be easier to read if the output from sha256sum was captured and the file was written manually? To me it seems like changing directories won't cover the case where the file isn't named in the expected way yet and it's not hard to tell the function what you want to have in there.

Reply:
I'm not necessarily worried about its legibility, more about why it's there. It's probably fine like thi.s

Reply:
To me it makes sense - what's being generated includes a strong reference to something else, so I'd rather be explicit about what that reference is instead of running the program in just the right way to produce the desired reference. If `sha256sum` offered a way to do that, I'd use that.

I think it would be less icky as a capture and write, though, so I think I'll do that instead of `awk`.